serum-390/soen390-team16-godzilla
# Production experimental

# Description

This PR: 
implements the production pipleline
enables creating production items for sales orders and canceling production items 
uses a logger to log production information in eventsLog.log file



## Link to work item

TODO: Include a link to the work item E.g.: #1

## How to Test
[use postman]
A ) TEST planning 
1) query orders
2) choose a sales order 
3) check inventory -> do the math for how many items[ final, raw]  will be left in inventory after production of order
4) plan production for that order using POST http://localhost:8080/api/production-manager/validate/?id= orderID &date=2020-12-3. NOTE [ if the entered date is before the current date the current date will be used by default] 
5) check inventory -> all the available items should be taken
6) check planned products db to see the production item using GET  http://localhost:8080/api/planning/
if bom items are missing the production status should be blocked. otherwise, it should be scheduled
7) check the orders db: a purchase order for the missing bom items should be created with the production id 
8) the flow is : after validating the order, if items are missing from bom -> create purchase order -> fire purchase order after 2 mins [ 2 min for demo purpose] -> the blocked production must be scheduled -> after 2 mins the production is complete [ production status complete , order status = ready]

B) TEST cancel planning 
1) query orders
2) choose a sales order 
3) check inventory -> do the math for how many items[ final, raw]  will be left in inventory after production of order
4) plan production for that order using POST http://localhost:8080/api/production-manager/validate/?id= orderID &date= someDate
5)  check planned products db to see the production item using GET  http://localhost:8080/api/planning/
6a) cancel planned product when status = blocked http://localhost:8080/api/production-manager/cancel//plannedProductID
6b) cancel planned product when status = scheduled http://localhost:8080/api/production-manager/cancel//plannedProductID
7) after canceling the status of planned product should be cancelled , the sales order should be reset to status = new, the items taken from inventory should be returned
